src/cljs_spa/router.cljs

```diff
@@ -36,10 +36,10 @@
 (defn create-router [routes]
   (let [name->route (->> routes (map (juxt :name identity)) (into {}))]
     (let [router (router5/createRouter (clj->js routes))]
-      (.usePlugin ^js/router5 router ((.-default router5-browser-plugin) #js{:useHash true}))
-      (.useMiddleware ^js/router5 router (fn [router]
-                                           (fn [to-state from-state]
-                                             (middleware* name->route to-state from-state)))))))
+      (.usePlugin ^js/Object router ((.-default router5-browser-plugin) #js{:useHash true}))
```

Comment:
I'm keeping an eye on this repo because you are doing very nice things and I'm learning new things :) So.. why to change a specific annotation to a generic one?

Reply:
Here's a question I posted recently:

While adapting cljs-spa-example for advanced compilation, I followed the Externs Guide and added a type annotation like so:

```(.usePlugin ^js/router5 router :somethin-else)```

This raises two questions: First, although adding this annotation works, I'm not actually sure that `js/router5` is the right type for this object. In fact AFAIK not every object in JavaScript has a class that you can point to.

Second, @bhauman mentioned that you can also use a generic annotation, `^js/Object`. In fact that works as well: https://github.com/pesterhazy/cljs-spa-example/pull/10/files Generic type hints aren't mentioned in the guide, however, so I'm wondering if there's any downside to just hinting interop sites with `^js/Object` across the board.

Reply:
The answer was that `^js` is sufficient for the annotation, so I'm going to go with that.